fastcrypto/src/merkle.rs

```diff
@@ -144,6 +155,142 @@ where
         }
         Some(current_hash)
     }
+
+    // Check if the proof is for the rightmost leaf in the tree
+    pub fn is_right_most(&self, leaf_index: usize) -> bool {
+        let mut level_index = leaf_index;
+        for sibling in self.path.iter() {
+            // The sibling hash of the current node
+            if level_index % 2 == 0 {
+                // The current node is a left child
+                if sibling.as_ref() != EMPTY_NODE.as_ref() {
+                    return false;
+                }
+            }
+            // Update to the level index one level up in the tree
+            level_index /= 2;
+        }
+        true
+    }
+}
+
+/// A proof that some data is not in a Merkle tree.
+#[derive(Serialize, Deserialize)]
+#[serde(bound(
+    serialize = "L: Serialize",
+    deserialize = "L: serde::de::DeserializeOwned"
+))]
+pub struct MerkleNonInclusionProof<L, T = Blake2b256>
```

Comment:
please document above this struct what is a non inclusion proof here (what is the proof, etc)